pyproject.toml

```diff
@@ -5,7 +5,6 @@ requires = [
     "Cython>=0.29.18",
     # We follow scipy for much of these pinnings
     # https://github.com/scipy/scipy/blob/master/pyproject.toml
-    "numpy==1.16.5; python_version=='3.6'",
     "numpy==1.16.5; python_version=='3.7'",
     "numpy==1.17.3; python_version=='3.8'",
```

Comment:
shouldn't we add 3.9 to this file? Not directly related to the PR of course.

Reply:
It is added below under a comment, I did wonder the same thing. 😊 I must say I don't understand why it's not pinned but I presume the scipy folks know what they're doing!

Reply:
you might as well pin to 1.19.4 since that is the only current version that has a build on pypi

Reply:
and leave it like that for pythyon 3.10

Reply:
There is a pin to 1.19.3 for 3.9 in https://github.com/scipy/oldest-supported-numpy. My understanding from the about section of that package is that we could just include `oldest-supported-numpy` within our pyproject.toml instead of maintaining all these special cases ourselves.

Reply:
would have to check with the scipy team about why we don't use it in SciPy itself, though

Reply:
There is an open PR at https://github.com/scipy/scipy/pull/13217 adding the Python 3.9 pin in SciPy itself. See also about pinning setuptools<=51.0 in that issue.

Reply:
Oldest supported numpy doesn't adhere to nep29. 

https://github.com/scipy/oldest-supported-numpy/issues/11

Reply:
Thanks, both. I don't understand the story with setuptools (is `<=51.0.0` necessary rather than `<=51`?), nor why they are only doing this on the 1.6 branch, but I've implemented the above-linked changes. Please merge or modify as you see fit...